src/Flow.jl

```diff
@@ -59,6 +59,21 @@ lowerBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop (
     Φ[I] = ϕuP(j,CIj(j,CI(I,i),N[j]-2),CI(I,i),u,ϕ(i,CI(I,j),u)) -ν*∂(j,CI(I,i),u); r[I,i] += Φ[I]) over I ∈ slice(N,2,j,2)
 upperBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop r[I-δ(j,I),i] -= Φ[CIj(j,I,2)] over I ∈ slice(N,N[j],j,2)
 
+"""
+    applyBodyForce!(r,t,g=(a,b)->0)
+
+This function apply body force `g` at time `t` to `r`, effectively the du/dt.
+Body force `g` input sould be function with arguments: index `a` and time `b`.
+The default of body force is null. 
+"""
+function applyBodyForce!(r,t,g=(a,b)->0)
+    N,n = size_u(r)
+    for i ∈ 1:n
+        gᵢ = g(i,t)
+        @loop r[I,i] += gᵢ over I ∈ inside(N)
+    end
+end
```

Comment:
If you think `g` will be expensive, then I guess you could do
```suggestion
applyBodyForce!(a::Flow{N},t) where N = for i in 1:N
    gᵢ = a.g(i,t)
    a.f[:,i] .+= gᵢ
end
```
Broadcasting is plenty in this kind of case.

Reply:
I will follow the idea of adding to all elements but I might still leave args the same since directly using `(a,t)` might limit the future usage. I can use `r::AbstractArray(T,Dv)` and use `Dv-1` to infer `N`.

Reply:
What are you envisioning for the more general usage of the function? A general body force routine would need to be a function of `x`, not just `t`, so we should use `apply!(a.f,(i,x)->BodyForce(i,x,t))`, right?

Reply:
Such a routine might be a good idea in the future (I _think_ that's a way we could get @b-fg 's SANS stuff integrated into the code) but this PR doesn't need to do everything everywhere all at once. ;-)